services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/TopicCleanupService.java

```diff
@@ -235,49 +238,48 @@ Admin getAdmin() {
+    // TODO - instead of adding all topics into the same queue we must simply use separate queues for RT and nonRT
+    // topics; this will save a lot of time wasted in sorting
     PriorityQueue<PubSubTopic> allTopics = new PriorityQueue<>(topicPriorityComparator);
     populateDeprecatedTopicQueue(allTopics);
     topicCleanupServiceStats.recordDeletableTopicsCount(allTopics.size());
     long refreshTime = System.currentTimeMillis();
+
     while (!allTopics.isEmpty()) {
       PubSubTopic topic = allTopics.poll();
+      String storeName = topic.getStoreName();
       try {
-        if (topic.isRealTime() && !multiDataCenterStoreToVersionTopicCount.isEmpty()) {
-          // Only delete realtime topic in child fabrics if all version topics are deleted in all child fabrics.
-          if (isRTTopicDeletionBlocked) {
+        if (topic.isRealTime() && (this.isRTTopicDeletionBlocked || !admin.isRTTopicDeletionPermittedByAllControllers(
+            admin.discoverCluster(storeName).getFirst(),
+            storeName,
+            null))) {
+          if (this.isRTTopicDeletionBlocked) {
             LOGGER.warn(
                 "Topic deletion for topic: {} is blocked due to unable to fetch version topic info",
                 topic.getName());
-            topicCleanupServiceStats.recordTopicDeletionError();
-            continue;
-          }
-          boolean canDelete = true;
-          for (Map.Entry<String, Map<String, Integer>> mapEntry: multiDataCenterStoreToVersionTopicCount.entrySet()) {
-            if (mapEntry.getValue().containsKey(topic.getStoreName())) {
-              canDelete = false;
-              LOGGER.info(
-                  "Topic deletion for topic: {} is delayed due to {} version topics found in datacenter {}",
-                  topic.getName(),
-                  mapEntry.getValue().get(topic.getStoreName()),
-                  mapEntry.getKey());
-              break;
-            }
-          }
-          if (!canDelete) {
-            continue;
+          } else {
+            LOGGER.warn("Topic deletion for topic: {} is delayed.", topic.getName());
           }
+          topicCleanupServiceStats.recordTopicDeletionError();
+          continue;
         }
-        getTopicManager().ensureTopicIsDeletedAndBlockWithRetry(topic);
-        topicCleanupServiceStats.recordTopicDeleted();
+      } catch (VeniceNoStoreException e) {
+        LOGGER.warn(
+            "Store {} not found. Exception when trying to delete topic: {} - {}",
+            storeName,
+            topic.getName(),
+            e.toString());
       } catch (VeniceException e) {
         LOGGER.warn("Caught exception when trying to delete topic: {} - {}", topic.getName(), e.toString());
         topicCleanupServiceStats.recordTopicDeletionError();
         // No op, will try again in the next cleanup cycle.
       }
-
+      topicCleanupServiceStats.recordTopicDeleted();
+      getTopicManager().ensureTopicIsDeletedAndBlockWithRetry(topic);
```

Comment:
is this right to move it out of the try-catch block and into the common path. This means that when exception happens, topic deleted will still be reported?